src/server/dflycmd.cc

```diff
@@ -366,6 +378,57 @@ void DflyCmd::ReplicaOffset(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void DflyCmd::ClusterGetSlotInfo(CmdArgList args, ConnectionContext* cntx) {
+  if (args.size() == 3) {
+    return (*cntx)->SendError(facade::WrongNumArgsError("DFLY CLUSTER GETSLOTINFO"),
+                              kSyntaxErrType);
+  }
+  ToUpper(&args[2]);
+  string_view slots_str = ArgS(args, 2);
+  if (slots_str != "SLOTS") {
+    return (*cntx)->SendError(kSyntaxErr, kSyntaxErrType);
+  }
+
+  vector<std::pair<SlotId, SlotStats>> slots_stats;
+  for (size_t i = 3; i < args.size(); ++i) {
+    string_view slot_str = ArgS(args, i);
+    uint32_t sid;
+    if (!absl::SimpleAtoi(slot_str, &sid)) {
+      return (*cntx)->SendError(kInvalidIntErr);
+    }
+    if (sid > ClusterConfig::kMaxSlotNum) {
+      return (*cntx)->SendError("Invalid slot id");
+    }
+    slots_stats.push_back(make_pair(sid, SlotStats{}));
+  }
+
+  Mutex mu;
+
+  auto cb = [&](auto*) {
+    EngineShard* shard = EngineShard::tlocal();
+    if (shard == nullptr)
+      return;
+
+    lock_guard lk(mu);
+    for (auto& slot_data : slots_stats) {
+      SlotStats stats = shard->db_slice().GetSlotStats(slot_data.first);
+      slot_data.second += stats;
+    }
+  };
+
+  shard_set->pool()->AwaitFiberOnAll(std::move(cb));
+
+  (*cntx)->StartArray(slots_stats.size());
+
+  for (const auto& slot_data : slots_stats) {
+    (*cntx)->StartArray(2);
+    (*cntx)->SendLong(slot_data.first);
+    (*cntx)->SendLong(slot_data.second.key_count);
```

Comment:
Perhaps it would be more useful to also include the name of this value (i.e. "key count"), as it could become confusing for users once we have >1 counters